ca/ca.go

```diff
@@ -125,7 +125,23 @@ func (h *Handle) QueueMultiple(it func(yield func(ar mtc.AssertionRequest) error
 	defer w.Close()
 	bw := bufio.NewWriter(w)
 
+	nextBatch := mtc.Batch{
+		Number: h.params.ActiveBatches(time.Now()).End + 1,
+		CA:     &h.params,
+	}
+	batchStart, batchEnd := nextBatch.ValidityInterval()
+
 	if err := it(func(ar mtc.AssertionRequest) error {
+		// Check that the assertion matches the checksum.
+		err = ar.Check()
+		if err != nil {
+			return err
+		}
+		// Clear the checksum as it will no longer be valid if NotAfter changes.
+		ar.Checksum = nil
```

Comment:
We could leave the notAfter field out of the checksum so we don't need to reset it, but if we're potentially going to add it to the Assertion later then it probably belongs. It's just a little weird for the checksums to be different before and after an assertion is queued, but ah well.

We could also reintroduce the `QueuedAssertion` struct as
```
type QueuedAssertion struct {
  AssertionRequest AssertionRequest
  NotAfter time.Time
}
```